src/olympia/landfill/serializers.py

```diff
@@ -103,25 +130,36 @@ def create_named_addon_with_author(self, name, author=None):
         if author is None:
             author = user_factory()
         try:
-            generate_user(author)
+            user = UserProfile.objects.create(username=author, email=f'{author}@email.com')
+            user.id = 4757633
         except Exception:  # django.db.utils.IntegrityError
             # If the user is already made, use that same user,
             # if not use created user
             addon = addon_factory(
-                status=STATUS_APPROVED,
+                status=amo.STATUS_APPROVED,
                 users=[UserProfile.objects.get(username=author)],
                 name=u'{}'.format(name),
                 slug=u'{}'.format(name),
+                version_kw={
+                'recommendation_approved': True,
+                'nomination': days_ago(6)
+                }
             )
             addon.save()
         else:
+            author.id = 4757633
```

Comment:
Same here, you will have to use author.update(id=settings.TASK_USER_ID)` ideally to not hardcode the id.